rm-datasource/src/main/java/io/seata/rm/datasource/undo/AbstractUndoLogManager.java

```diff
@@ -392,5 +414,18 @@ protected abstract void insertUndoLogWithNormal(String xid, long branchId, Strin
      * @return
      * @throws SQLException
      */
-    protected abstract byte[] getRollbackInfo(ResultSet rs) throws SQLException;
+    protected byte[] getRollbackInfo(ResultSet rs) throws SQLException  {
+        byte[] rollbackInfo = rs.getBytes(ClientTableColumnsName.UNDO_LOG_ROLLBACK_INFO);
+        CompressorType compressType = CompressorType.getByCode(rs.getInt(ClientTableColumnsName.UNDO_LOG_COMPRESS_TYPE));
+        return CompressorFactory.getCompressor(compressType.getCode()).decompress(rollbackInfo);
```

Comment:
rollbackInfo needs to determine whether to decompress according to the compression type value in the context.

Reply:
ok, move to context